packages/deployments/contracts/contracts/messaging/connectors/axelar/BaseAxelar.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.17;
+
+import {AxelarExecutable} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/executable/AxelarExecutable.sol";
+import {IAxelarGateway} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGateway.sol";
+import {IAxelarGasService} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGasService.sol";
+import {StringToAddress, AddressToString} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/AddressString.sol";
+
+abstract contract BaseAxelar is AxelarExecutable {
+  using StringToAddress for string;
+  using AddressToString for address;
+
+  // ============ Internal Storage ============
+  IAxelarGasService public immutable gasService;
+
+  // Mirror chain id
+  string public MIRROR_CHAIN_ID;
+
+  // ============ Constructor ============
+  constructor(address _amb, bytes memory _axelarParams) AxelarExecutable(_amb) {
+    (address _gasReceiver, string memory _mirrorChainId) = abi.decode(_axelarParams, (address, string));
+    // sanity checks
+    require(bytes(_mirrorChainId).length != 0, "!mirrorChainId");
+
+    // set immutable propertioes
+    gasService = IAxelarGasService(_gasReceiver);
+    MIRROR_CHAIN_ID = _mirrorChainId;
+  }
+
+  // ============ Public Fns ============
+  function _execute(
+    string calldata _sourceChain,
+    string calldata _sourceAddress,
```

Comment:
should `_verifySender` be implemented at this layer if its the same to check the crosschain sender on both hub and spoke?

Reply:
isn't that passed into the function?